### ISSUE TITLE
Exception in server tick loop

### ISSUE THREAD

**Issue by Zoidey** [ldtteam/minecolonies#755]
Updated to the latest version and the version before (originally) and the server is crashing. here is part of the log if you what or need the full blown version I'll send that but didn't want to spam it here:

Time: 4/3/17 6:24 AM
Description: Exception in server tick loop

java.lang.NullPointerException: Exception in server tick loop
	at com.minecolonies.coremod.colony.workorders.WorkOrderBuild.<init>(WorkOrderBuild.java:77)
	at com.minecolonies.coremod.entity.ai.citizen.builder.ConstructionTapeHelper.removeConstructionTape(ConstructionTapeHelper.java:213)
	at com.minecolonies.coremod.colony.buildings.AbstractBuilding.onDestroyed(AbstractBuilding.java:550)
	at com.minecolonies.coremod.colony.buildings.BuildingHome.onDestroyed(BuildingHome.java:90)
	at com.minecolonies.coremod.colony.buildings.AbstractBuilding.destroy(AbstractBuilding.java:531)
	at com.minecolonies.coremod.colony.Colony$$Lambda$1130/1988602137.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.minecolonies.coremod.colony.Colony.cleanUpBuildings(Colony.java:1147)
	at com.minecolonies.coremod.colony.Colony.onWorldTick(Colony.java:1072)
	at com.minecolonies.coremod.colony.ColonyManager.lambda$onWorldTick$46(ColonyManager.java:653)
	at com.minecolonies.coremod.colony.ColonyManager$$Lambda$1127/476042743.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.minecolonies.coremod.colony.ColonyManager.onWorldTick(ColonyManager.java:653)
	at com.minecolonies.coremod.event.FMLEventHandler.onWorldTick(FMLEventHandler.java:50)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_761_FMLEventHandler_onWorldTick_WorldTickEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
	at net.minecraftforge.fml.common.FMLCommonHandler.onPreWorldTick(FMLCommonHandler.java:282)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:694)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:387)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471)
	at java.lang.Thread.run(Thread.java:745)

***Note* I went back to Minecolonies v0.7.3 and server started without any problems.**

**Comment by Raycoms** [comment]
Can you check if this here solves it:
https://www.dropbox.com/sh/9cnqcjl0nan57cy/AAD8GxTwBtxSLLEZ9wQfiE0sa/branch/fix/construction-tape?dl=0&preview=minecolonies-universal-1.10.2-0.6.3331.jar

2017-04-03 3:30 GMT-03:00 Zoidey <notifications@github.com>:

**Comment by Zoidey** [comment]
I managed to get it going after going back a few versions and re-updating my server. I've came across another couple of issues that I'll start a new topic for (wanna try and test something first). If I have that issue again I'll certainly try your suggestion and let you know how I got on.

**Comment by Raycoms** [comment]
The jar I just sent has all the newest changes, should fix the issue you
had and might also have some fixes for other problems.

2017-04-03 12:41 GMT-03:00 Zoidey <notifications@github.com>:

**Comment by Zoidey** [comment]
Oh okay goody. The issues I'm having are:

I was not able to interact with some buildings (I rebuild them and upgraded the fixed them)

I had to rebuild a miner's hut and its fully upgraded but the miner keeps asking me to update the miners hut so they can dig deeper (they cant dig as there are no levels for me to select them to dig). I thought it might be because I have to rebuild some of the houses and upgrade them again to mac (currently doing this but its taking awhile as I got too many houses.... got a tad carried away awhile ago). I'll Also note that I re hired the exact same miner as well as a completely different miner and still no luck.

I'll give the new jar a shot and let you know how I get on thanks.

**Comment by Raycoms** [comment]
You used the jar I sent you? That should resolve the problem with the
having to rebuild.
Yeah you will have to remove the miner completely, including his ladders
and then rebuild him.

2017-04-03 20:24 GMT-03:00 Zoidey <notifications@github.com>:

**Comment by Zoidey** [comment]
Using it now, just started the server so I'll have let you know in a minute or two :)

Oh never thought to remove the ladders, thanks.

*Note 1* Trying to rebuild the miner and I cant even interact with it to build it again.

*Note 2* Okay disconnecting and reconnecting fixes it, I can interact by right clicking now :)

*Note 3* I got the miner to work again (I think), I rebuilt it and rehired the miner and it started mining but stopped and wouldn't do anything else so I rebuild once again and fully upgraded first and then rehired the miner after the upgrade and that seems to have fixed it as they are digging the main shaft pass the point they stopped. I think I might know why the miner stopped, they needed cobblestone to fill in the shaft to rebuild it but they don't seem to be requesting cobblestone like they would if they needed torches, ladders, fences or oak planks etc. Maybe its a bug or an option that could be added/ fixed?
